src/utils/userLocation.ts

```diff
@@ -1,81 +1,129 @@
-import firebase, { UserInfo } from "firebase/app";
-
-import { Room } from "types/rooms";
+import firebase from "firebase/app";
 
 import { updateUserProfile } from "pages/Account/helpers";
 import { useInterval } from "hooks/useInterval";
 
+import { LOCATION_INCREMENT_MS, LOCATION_INCREMENT_SECONDS } from "settings";
+
 import { getCurrentTimeInMilliseconds } from "./time";
-import { openRoomUrl, openUrl, venueInsideUrl } from "./url";
+import { openRoomUrl } from "./url";
 
-const LOCATION_INCREMENT_SECONDS = 10;
-const LOCATION_INCREMENT_MS = LOCATION_INCREMENT_SECONDS * 1000;
+export type LocationData = Record<string, number>;
 
-type LocationData = Record<string, number>;
+export type UpdateLocationDataProps = {
+  userId: string;
+  newLocationData: LocationData;
+};
 
-export const updateLocationData = (
-  userId: string,
-  newLocationData: LocationData = {}
-) => {
+export const updateLocationData = ({
+  userId,
+  newLocationData,
+}: UpdateLocationDataProps) => {
   updateUserProfile(userId, {
     lastSeenAt: getCurrentTimeInMilliseconds(),
     lastSeenIn: newLocationData,
   });
 };
 
-export interface TrackLocationProps {
+export type SetLocationDataProps = {
   userId: string;
   locationName: string;
-  lastSeenIn?: LocationData;
-}
+};
 
-export const trackLocationEntered = ({
+export const setLocationData = ({
   userId,
   locationName,
-}: TrackLocationProps) => {
-  updateLocationData(userId, {
-    [locationName]: getCurrentTimeInMilliseconds(),
+}: SetLocationDataProps) => {
+  updateLocationData({
+    userId,
+    newLocationData: {
+      [locationName]: getCurrentTimeInMilliseconds(),
+    },
+  });
+};
+
+export type UpdateCurrentLocationDataProps = {
+  userId: string;
+  profileLocationData: LocationData;
+};
+
+export const updateCurrentLocationData = ({
+  userId,
+  profileLocationData,
+}: UpdateCurrentLocationDataProps) => {
+  const [locationName] = Object.keys(profileLocationData);
+
+  updateLocationData({
+    userId,
+    newLocationData: { [locationName]: getCurrentTimeInMilliseconds() },
   });
 };
 
-export const enterVenue = (venueId: string) => openUrl(venueInsideUrl(venueId));
+export const clearLocationData = (userId: string) => {
+  updateLocationData({ userId, newLocationData: {} });
+};
 
-type EnterExternalRoomProps = {
+export type EnterExternalRoomProps = {
   userId: string;
-  room: Room;
+  roomUrl: string;
+  roomId: string;
 };
 
-export const enterExternalRoom = ({ userId, room }: EnterExternalRoomProps) => {
-  trackLocationEntered({
+export const enterExternalRoom = ({
+  userId,
+  roomId,
+  roomUrl,
+}: EnterExternalRoomProps) => {
+  setLocationData({
     userId,
-    locationName: room.url,
+    locationName: roomId,
   });
 
-  openRoomUrl(room.url);
+  openRoomUrl(roomUrl);
+};
+
+export type UseUpdateTimespentPeriodicallyProps = {
```

Comment:
As per https://github.com/sparkletown/sparkle/pull/1062#issuecomment-772978074 this should use `interface` rather than `type`

Reply:
https://github.com/sparkletown/sparkle/pull/1129#discussion_r570165378

Reply:
fixed in 200c94f53638642cd3f70418f66b2accb8f392fe